utils/check_repo.py

```diff
@@ -1063,8 +1062,41 @@ def check_all_objects_are_documented():
             "The following objects are in the public init so should be documented:\n - "
             + "\n - ".join(undocumented_objs)
         )
-    check_docstrings_are_in_md()
     check_model_type_doc_match()
+    check_public_method_exists(documented_methods_map)
+
+
+def check_public_method_exists(documented_methods_map):
+    """Check that all explicitly documented public methods are defined in the corresponding class."""
+    failures = []
+    for obj, methods in documented_methods_map.items():
+        # Let's ensure there is no repetition
+        if len(set(methods)) != len(methods):
+            failures.append(f"Error in the documentation of {obj}: there are repeated documented methods.")
+
+        # Navigates into the object, given the full import path
+        nested_path = obj.split(".")
+        submodule = transformers
+        if len(nested_path) > 1:
+            nested_submodules = nested_path[:-1]
+            for submodule_name in nested_submodules:
+                if submodule_name == "transformers":
+                    continue
+                submodule = getattr(submodule, submodule_name)
+        class_name = nested_path[-1]
+        obj_class = getattr(submodule, class_name)
+        # Checks that all explicitly documented methods are defined in the class
+        for method in methods:
+            if method == "all":  # Special keyword to document all public methods
+                continue
+            if not hasattr(obj_class, method):
+                failures.append(
+                    "The following public methods are explicitly documented but not defined in the corresponding "
+                    f"class: {obj}.{method}"
+                )
```

Comment:
Sanity check: commenting out publicly documented functions triggers this failure